app/models/programme_activity_grouping.rb

```diff
@@ -58,6 +58,34 @@ def users_completed(users:)
     end.to_h
   end
 
+  # Move "Implement selected key stage 3 Teach Computing Curriculum resources" from section 2
+  # to section 3 and update the #required_for_completion value on the old grouping
+  def self.move_i_belong_section_from_2_to_3
+    # Perform the move of the activity to the new grouping
+    programme = Programme.find_by(slug: "i-belong")
+    raise "Programme could not be found" if programme.blank?
+
+    potential_groupings = programme.programme_activity_groupings.community
+    raise "There are more than 2 available groupings for the moved activity" unless potential_groupings.size == 2
+
+    activity = programme.activities.find_by(title: "Implement selected key stage 3 Teach Computing Curriculum resources")
+    raise "Activity could not be found" if activity.blank?
+
+    programme_activity = ProgrammeActivity.find_by(activity_id: activity.id)
+    raise "Programme activity could not be found" if programme_activity.blank?
+
+    original_grouping = programme_activity.programme_activity_grouping
+    raise "Current grouping could not be found" if original_grouping.blank?
+
+    new_grouping = potential_groupings.where.not(id: original_grouping.id).first
+    raise "New grouping could not be determined" if new_grouping.blank?
+
+    programme_activity.update!(programme_activity_grouping_id: new_grouping.id)
+
+    # Update the original grouping to only require 2 completed activities
+    original_grouping.update!(required_for_completion: 2)
+  end
```

Comment:
Running the method twice flipped it back, but both groups then had required_for_completion set to 2. Might need something to catch this, else can run into an outcome of updating the last group to "Increase girls' engagement by completing at least two activities".

Reply:
Nice, instead of a guard I just ensured that it would run the same every time in the end!

Reply:
Just re-ran the method, I don't think it has had quite the desired affect, as 'Implement selected key stage 3 Teach Computing Curriculum resources' isn't visible in either group for me afterwards (although the "at least one activity" has remained!).

Reply:
Should work OK now

Reply:
Yep, that looks to have addressed it - nice one. 👍